src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -0,0 +1,158 @@
+import org.apache.pdfbox.multipdf.Splitter;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDDocumentInformation;
+import org.apache.pdfbox.text.PDFTextStripper;
+import org.apache.tomcat.util.http.fileupload.IOUtils;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.multipart.commons.CommonsMultipartFile;
+
+import de.tum.in.www1.artemis.domain.Attachment;
+import de.tum.in.www1.artemis.domain.enumeration.AttachmentType;
+import de.tum.in.www1.artemis.domain.lecture.AttachmentUnit;
+import de.tum.in.www1.artemis.service.util.Tuple;
+import de.tum.in.www1.artemis.web.rest.dto.LectureUnitDTO;
+import de.tum.in.www1.artemis.web.rest.dto.LectureUnitInformationDTO;
+import de.tum.in.www1.artemis.web.rest.dto.LectureUnitSplitDTO;
+
+@Service
+public class LectureUnitProcessingService {
+
+    /**
+     * Split units from given file according to given split information.
+     * @param lectureUnitSplitDTOs The split information
+     * @param file The file (lecture slide) to be split
+     * @return The prepared units to be saved
+     */
+    public List<LectureUnitDTO> splitUnits(List<LectureUnitSplitDTO> lectureUnitSplitDTOs, MultipartFile file) throws IOException {
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        List<LectureUnitDTO> units = new ArrayList<>();
+        Splitter pdfSplitter = new Splitter();
+        PDDocument document = PDDocument.load(file.getBytes());
+
+        for (LectureUnitSplitDTO lectureUnit : lectureUnitSplitDTOs) {
+            AttachmentUnit attachmentUnit = new AttachmentUnit();
+            Attachment attachment = new Attachment();
+            PDDocumentInformation pdDocumentInformation = new PDDocumentInformation();
+
+            pdfSplitter.setStartPage(lectureUnit.startPage());
+            pdfSplitter.setEndPage(lectureUnit.endPage());
+            pdfSplitter.setSplitAtPage(lectureUnit.endPage());
+
+            List<PDDocument> documentUnits = pdfSplitter.split(document);
+            pdDocumentInformation.setTitle(lectureUnit.unitName());
+            documentUnits.get(0).setDocumentInformation(pdDocumentInformation);
+            documentUnits.get(0).save(outputStream);
+
+            // setup attachmentUnit and attachment
+            attachmentUnit.setDescription("");
+            attachment.setName(lectureUnit.unitName());
+            attachment.setAttachmentType(AttachmentType.FILE);
+            attachment.setVersion(1);
+            attachment.setReleaseDate(lectureUnit.releaseDate());
+            attachment.setUploadDate(ZonedDateTime.now());
+
+            // prepare file to be set from byte[] to MultipartFile by using CommonsMultipartFile
+            String tempDirectory = System.getProperty("java.io.tmpdir");
+            File tempFile = new File(tempDirectory, lectureUnit.unitName() + ".pdf");
```

Comment:
Please use `Path.of` instead of `new File()`, especially when you call `toPath()` later anyways.